app/src/main/java/com/google/droidjump/GameView.java

```diff
@@ -175,10 +183,53 @@ private void updateObstaclesCoordinates() {
             if (obstacle.getX() + obstacle.getWidth() < 0) {
                 it.remove();
                 score += 10;
+                if (activity.getSavedSignedInAccount() != null) {
+                    if (obstacle instanceof Cactus) {
+                        cactusScore += 1;
+                    } else if (obstacle instanceof Bat) {
+                        batScore += 1;
+                    } else if (obstacle instanceof Palm) {
+                        palmScore += 1;
+                    }
+                }
             }
         }
     }
 
+    private void updateObstacleLeaderboards() {
+        if (activity.getSavedSignedInAccount() == null) {
+            return;
+        }
+        LeaderboardsClient client = activity.getLeaderboardsClient();
+        int[] leaderboards = {R.string.leaderboard_cactus_jumper, R.string.leaderboard_bat_avoider, R.string.leaderboard_palm_climber};
+        Resources resources = getResources();
+        for (int leaderboard : leaderboards) {
+            String leaderboardId = resources.getString(leaderboard);
+            client.loadCurrentPlayerLeaderboardScore(
+                    leaderboardId,
+                    LeaderboardVariant.TIME_SPAN_ALL_TIME,
+                    LeaderboardVariant.COLLECTION_PUBLIC).addOnSuccessListener(data -> {
+                        LeaderboardScore leaderboardScore = data.get();
+                        long newScore;
+                        switch (leaderboard) {
+                            case R.string.leaderboard_cactus_jumper:
+                                newScore = cactusScore;
+                                break;
+                            case R.string.leaderboard_bat_avoider:
+                                newScore = batScore;
+                                break;
+                            default:
```

Comment:
Still create a case for "leaderboard_palm_climber". "default" would mean that something went wrong.